src/ui/ui_nav_link_collection.js

```diff
@@ -26,4 +26,9 @@ export default class UiNavLinkCollection extends Collection {
     return this[inOrderCache];
   }
 
+  delete(value) {
+    super.delete(value);
+    this[inOrderCache] = null;
```

Comment:
How about moving this before `super.delete()` and returning the calls return value?